app/search/README.md

```diff
@@ -0,0 +1,123 @@
+# The Mampf Search Architecture
```

Comment:
I added this README since I thought it might be helpful in the future. Please check it in detail since there has been some time between writing the code and writing this README. I might have forgotten or misremembered something.

Reply:
Nice, I've reworked it a bit in ac757a611908. Especially, I removed sections that talked about the general philosophy, e.g. single-responsibility etc. since that is true for the entire codebase. Furthermore, I removed duplicated explanations.